meta-lxatac-software: labgrid: allow overriding USB ports in userconfig

hnez opened the pull request:
This should make it easier to attach USB hubs to the TAC and getting the matches right.

I'm not too experienced with jinja2 so I've had some trouble working around the scoping rules (`set`ing a variable in a `include`d file does not have an effect on the file that did the `include`ing, but modifing the variable in another way does).
And preventing the output from my `usb_ports.pop` and `.extend` tricks from being rendered (that's what the `and '' and `or ''` are for).
Maybe you have some better idea.

@a3f could this be of any use to you?

hnez committed:
meta-lxatac-software: labgrid: allow overriding jinja2 variables in userconfig

Set up some variables in configuration.yaml using sane defaults, then include
userconfig.yaml and allow overriding the defaults from there, _then_ set up
ressources based on said variables.

This should make it easier to attach USB hubs to the TAC and getting the
matches right and sets up a pattern that we can use later to add features.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

jluebbe reviewed (changes requested):
Looks good otherwise.

hnez committed:
meta-lxatac-software: labgrid: deploy a default userconfig.yaml

Otherwise users may get out of touch with the most recent defaults due to
the userconfig.yaml being migrated from rauc install to rauc install.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>
